src/masoniteorm/relationships/HasManyThrough.py

```diff
@@ -57,33 +57,46 @@ def __get__(self, instance, owner):
             if attribute in instance._relationships:
                 return instance._relationships[attribute]
 
-            result = self.apply_query(
+            result = self.apply_related_query(
```

Comment:
why is this renamed?

Reply:
@josephmancuso 

this was renamed as the BaseRelationship methos has a completely different sigbature and parameter requirements
from BaseRelationship on line 72
```python
def apply_query(self, foreign, owner, foreign_key, local_key):
```
Though the usage of this method is similar the parameter types are entirely different and the number of required parameters is different too.
Renaming the method means it will not be used as a subclass overide in future.

Reply:
@josephmancuso 
Based on this I had a look at all the other relationships and all of the (except HasOneThrough)
have method signatures for `apply_query) that do not match the BaseRelationship method signature.

Just thinking uot loud here... 
We should probably look at aligning these classes with the BaseRelationship in a separate PR